### PR TITLE
code cleanup: move parameter to aggregate, simplify code

### PR DESCRIPTION
**Pull request by hboutemy**

### CODE REVIEW

File: src/main/java/org/cyclonedx/maven/BaseCycloneDxMojo.java [hunk 1]
```diff
@@ -176,30 +171,6 @@ public abstract class BaseCycloneDxMojo extends AbstractMojo implements Contextu
     @Parameter(property = "excludeTypes", required = false)
     private String[] excludeTypes;
 
-    /**
-     * Excluded ArtifactIds.
-     */
-    @Parameter(property = "excludeArtifactId", required = false)
-    protected String[] excludeArtifactId;
```
Comment (stevespringett) [inline src/main/java/org/cyclonedx/maven/BaseCycloneDxMojo.java:183]: This should continue to be in BaseCycloneDxMojo. Both makeBom and makeAggregateBom should account for exclusions.

File: src/main/java/org/cyclonedx/maven/BaseCycloneDxMojo.java [hunk 2]
```diff
@@ -176,30 +171,6 @@ public abstract class BaseCycloneDxMojo extends AbstractMojo implements Contextu
...
-
-    /**
-     * Excluded GroupIds.
-     */
-    @Parameter(property = "excludeGroupId", required = false)
-    protected String[] excludeGroupId;
```
Comment (stevespringett) [inline src/main/java/org/cyclonedx/maven/BaseCycloneDxMojo.java:189]: This should continue to be in BaseCycloneDxMojo. Both makeBom and makeAggregateBom should account for exclusions.

**Review by stevespringett — CHANGES_REQUESTED** [review]
Thanks for the cleanup. Looks good. Just a few minor changes requested.

File: src/main/java/org/cyclonedx/maven/BaseCycloneDxMojo.java (hunk 1 above)
Comment (hboutemy) [reply src/main/java/org/cyclonedx/maven/BaseCycloneDxMojo.java:183]: it currently is not taken into account in makeBom, or the refactoring would not have been ok: IIUC, this means there is a bug in makeBom, that will require another fix.
I'll revert that part of the refactoring in the current PR, and open another bugfix PR for taking the config into account into makeBom

**Comment by hboutemy** [comment]
on `excludeGroupId` and `excludeArtifactId`, I reverted the refactoring as requested and starting investigation.

First finding: the feature was added in #91 with `excludeArtifactId` and `excludeTestProject` and completed with `excludeGroupId` in #218 with documentation in README focusing on `makeAggregateBom`

If we extend the usage, we'll need to update documentation...

File: src/main/java/org/cyclonedx/maven/BaseCycloneDxMojo.java (hunk 2 above)
Comment (hboutemy) [reply src/main/java/org/cyclonedx/maven/BaseCycloneDxMojo.java:189]: moved back to BaseCycloneDxMojo

**Comment by stevespringett** [comment]
Big thanks @hboutemy

**Pull request merged**